src/validation.cpp

```diff
@@ -3656,8 +3663,8 @@ CBlockIndex * CChainState::InsertBlockIndex(const uint256& hash)
 
 bool CChainState::LoadBlockIndex(const Consensus::Params& consensus_params, CBlockTreeDB& blocktree)
 {
+    LOCK(cs_main);
     if (!blocktree.LoadBlockIndexGuts(consensus_params, [this](const uint256& hash){ return this->InsertBlockIndex(hash); }))
-        return false;
 
```

Comment:
Dropping the "return false" doesn't seem right...

Reply:
Ouch, thanks! Will fix!